indexer-app/src/cmr/indexer/services/index_service.clj

```diff
@@ -425,6 +426,22 @@
   [context concept]
   (meta-db/get-associations-for-tool context concept))
 
+(defn create-association-map-keys
+  "Creates the association map keys that lists out all of the different association types.
+  For example, this function changes the named concept type from variable-association to a 
+  key of :variable-associations."
+  [key-str]
+  {key-str (keyword (str key-str "s")) })
+
+(defn get-associations
+  "Get all of the associations for the passed in concept. Group the results by concept type
+   so that it aligns with what index-concept :default expects."
+  [context concept]
+  (let [combined-assocs (meta-db/find-associations context concept)
+        grouped-assocs (group-by :concept-type combined-assocs)
+        new-key-map (into {} (map #(create-association-map-keys %) (keys grouped-assocs)))]
+    (cset/rename-keys grouped-assocs new-key-map)))
```

Comment:
Rename the keys to keyword to be able to get the values out in a certain way (e.g. `(:tag-associations associations)` below) is unnecessary, especially for a PR that addresses performance. We can get the values out without renaming the keys (e.g. `(get associations "tag-association")`). This will make the code easier to read and more performant.

Reply:
Originally, I thought the same, but then I didn't want to have to change a lot of namespaces down the line, namely the search results that uses the groupings. After I looked I think I can make the change without having to change a bunch of code down the line.

Reply:
So scratch what I said.  Basically I tried to keep it the same in this function, because bulk indexing uses different functions but also uses the same keys as does indexing deleting of concepts. Also there are two sets of get-elastic-version code, one in the index_service namespace and the other in the elasticsearch namespace that also use the same keys.  I will be happy to create a ticket to address this request so that it will work everywhere in January. I only have one more day to get this out and approved and tested in SIT this week.  After that I am out until January.

Reply:
Since you are also out, I will write a ticket to do the work in a few weeks.